client/planning-extension/src/authoring-react-fields/location/editor.tsx

```diff
@@ -1,34 +1,50 @@
 import * as React from 'react';
 import {IEditorComponentProps} from 'superdesk-api';
 import {ILocationFieldConfig, ILocationFieldUserPreferences, ILocationValueOperational} from './interfaces';
+import {ILocation} from '../../../../interfaces';
 import {extensionBridge} from '../../extension_bridge';
+import {cloneDeep, set} from 'lodash';
 
 type IProps = IEditorComponentProps<ILocationValueOperational, ILocationFieldConfig, ILocationFieldUserPreferences>;
+type LocationOrDetailsValue = ILocation | string | null;
+
+const isLocationObject = (v: LocationOrDetailsValue): v is ILocation =>
+    typeof v === 'object' && v !== null;
 
 export class Editor extends React.PureComponent<IProps> {
     render() {
         const {EditorFieldLocation} = extensionBridge.editor.fields;
+        const location: ILocation | null = this.props.value?.[0] ?? this.props.value;
```

Comment:
when doing something special/unexpected like conditionally getting first item or the entire list, add a comment clarifying what's going on. Maybe default isn't a list at all?